projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+	case ADC_SINGLE_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch3 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch4 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch5 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, IDLE_MODE);
+		break;
+	case ADC_CONTINUOUS_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch3 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch4 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch5 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, CONTINUOUS_CONV);
+		break;
+	default:
+		ret = -1;
+		break;
+	}
+
+	return ret;
+}
+
+/**
+ * CN0411 read temperature
+ *
+ * Reads channel 0 of the ADC
+ * Based on the conversion mode used:
+ * 	- continuous: since the function is called by the user and the sequence
+ * 	  of channels conversions cannot be controlled individually, the function
+ * 	  assures that the conversion data is read properly, gathering the data only
+ * 	  after the transition is made on the desired ADC channel. Therefore, if the
+ * 	  currently converted ADC channel is channel 0, the	program waits until next
+ * 	  channel conversion is made and reads the data register when transition to
+ * 	  channel 0 is detected.
+ * 	- single: read channel 0 individually
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_read_temp(cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+	uint32_t ch0_data = 0, status_reg;
+
+	switch (cn0411_dev->conv_type) {
+	case ADC_CONTINUOUS_CONV:
+		ret = AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+					  &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+		status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+			     & ADC_CH_RDY_MSK;
+
+		while(status_reg == ADC_CH0) {
+			ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+						   &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+			status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+				     & ADC_CH_RDY_MSK;
+		}
+		while(status_reg != ADC_CH0) {
+			ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+						   &cn0411_dev->ad7124_dev.regs[AD7124_Status]);
+			status_reg = cn0411_dev->ad7124_dev.regs[AD7124_Status].value
+				     & ADC_CH_RDY_MSK;
+		}
+		ret |= AD7124_ReadRegister(&cn0411_dev->ad7124_dev,
+					   &cn0411_dev->ad7124_dev.regs[AD7124_Data]);
+		ch0_data = cn0411_dev->ad7124_dev.regs[AD7124_Data].value;
+		break;
+	case ADC_SINGLE_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, SINGLE_CONV);
+		if (AD7124_WaitForConvReady(&cn0411_dev->ad7124_dev, ADC_TIMEOUT)
+		    == -3) {
+			printf("TIMEOUT\n");
```

Comment:
Try to fit this in one line:
```
if (AD7124_WaitForConvReady(&cn0411_dev->ad7124_dev, ADC_TIMEOUT) == -3)
        printf("TIMEOUT\n");
```
Also, what does '-3' mean? Use a define or enum instead